### PR TITLE
feat(RHTAPREL-653): update pipeline to use json path from task result

### PR DESCRIPTION
**Pull request by happybhati**
- modified the pipeline to dynamically source the `data.json`,
  `snapshot_spec.json` and `release_plan_admission` files from
  the results of the `collect-data` task.

### CODE REVIEW

**Review by johnbieren — CHANGES_REQUESTED** [review]
This should also include the other CR json paths

**Comment by happybhati** [comment]
> This should also include the other CR json paths

added thanks for review :)

**Review by mmalina — DISMISSED** [review]

File: pipelines/e2e/README.md [hunk 1]
```diff
@@ -18,6 +18,10 @@ affected by RHTAP services or which results could affect the RHTAP workflow.
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 1.9.0
```
Comment (johnbieren) [inline pipelines/e2e/README.md]: Wrong version

File: pipelines/deploy-release/README.md [hunk 2]
```diff
@@ -16,6 +16,10 @@ Tekton pipeline to verify Snapshot prior to Deployment
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 0.14.0
+- modified the pipeline to dynamically source the `snapshot_spec.json` file
```
Comment (johnbieren) [inline pipelines/deploy-release/README.md]: Capital M to match the other entries please

File: pipelines/e2e/README.md [hunk 3]
```diff
@@ -18,6 +18,10 @@ affected by RHTAP services or which results could affect the RHTAP workflow.
...
+- modified the pipeline to dynamically source the `snapshot_spec.json`
```
Comment (johnbieren) [inline pipelines/e2e/README.md]: Capital M and `s/-/*/` to match the other entries please

File: pipelines/push-to-external-registry/README.md [hunk 4]
```diff
@@ -17,6 +17,10 @@ Tekton pipeline to release Snapshots to an external registry.
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 2.2.0
+- modified the pipeline to dynamically source the `data.json`, `snapshot_spec.json` and
```
Comment (johnbieren) [inline pipelines/push-to-external-registry/README.md]: Capital M to match the other entries please

File: pipelines/push-to-external-registry/README.md [hunk 5]
```diff
@@ -17,6 +17,10 @@ Tekton pipeline to release Snapshots to an external registry.
...
+  `release_plan_admission` files from the results of the `collect-data` task.
```
Comment (johnbieren) [inline pipelines/push-to-external-registry/README.md]: Why do data.json and snapshot_spec.json have the .json suffix and not release_plan_admission?

File: pipelines/release-to-github/README.md [hunk 6]
```diff
@@ -17,6 +17,10 @@ Tekton release pipeline to release binaries extracted from the image built with
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 1.2.0
+- modified the pipeline to dynamically source the `snapshot_spec.json`
```
Comment (johnbieren) [inline pipelines/release-to-github/README.md]: Capital M to match the other entries please

File: pipelines/release-to-github/README.md [hunk 7]
```diff
@@ -17,6 +17,10 @@ Tekton release pipeline to release binaries extracted from the image built with
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 1.2.0
```
Comment (johnbieren) [inline pipelines/release-to-github/README.md]: One too many #

File: pipelines/rh-push-to-external-registry/README.md [hunk 8]
```diff
@@ -17,6 +17,10 @@ Tekton pipeline to release Red Hat Snapshots to an external registry. This pipel
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 2.2.0
+- modified the pipeline to dynamically source the `data.json`, `snapshot_spec.json` and
```
Comment (johnbieren) [inline pipelines/rh-push-to-external-registry/README.md]: Capital M to match the other entries please

File: pipelines/rh-push-to-external-registry/README.md [hunk 9]
```diff
@@ -17,6 +17,10 @@ Tekton pipeline to release Red Hat Snapshots to an external registry. This pipel
...
+  `release_plan_admission` files from the results of the `collect-data` task.
```
Comment (johnbieren) [inline pipelines/rh-push-to-external-registry/README.md]: Same comment about .json suffix

File: pipelines/rh-push-to-registry-redhat-io/README.md [hunk 10]
```diff
@@ -17,6 +17,10 @@ Tekton pipeline to release content to registry.redhat.io registry.
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 1.9.0
+- modified the pipeline to dynamically source the `data.json`, `snapshot_spec.json` and
```
Comment (johnbieren) [inline pipelines/rh-push-to-registry-redhat-io/README.md]: Capital M and `s/-/*/` to match the other entries please

File: pipelines/rh-push-to-registry-redhat-io/README.md [hunk 11]
```diff
@@ -17,6 +17,10 @@ Tekton pipeline to release content to registry.redhat.io registry.
...
+  `release_plan_admission` files from the results of the `collect-data` task.
```
Comment (johnbieren) [inline pipelines/rh-push-to-registry-redhat-io/README.md]: .json

File: pipelines/rh-push-to-registry-redhat-io/README.md [hunk 12]
```diff
@@ -17,6 +17,10 @@ Tekton pipeline to release content to registry.redhat.io registry.
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 1.9.0
```
Comment (johnbieren) [inline pipelines/rh-push-to-registry-redhat-io/README.md]: One too many #

File: pipelines/rhtap-service-push/README.md [hunk 13]
```diff
@@ -19,6 +19,10 @@
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 1.5.0
```
Comment (johnbieren) [inline pipelines/rhtap-service-push/README.md]: One too many #

File: pipelines/rhtap-service-push/README.md [hunk 14]
```diff
@@ -19,6 +19,10 @@
...
+- modified the pipeline to dynamically source the `data.json`, `snapshot_spec.json` and
```
Comment (johnbieren) [inline pipelines/rhtap-service-push/README.md]: Capital M to match the other entries please

File: pipelines/rhtap-service-push/README.md [hunk 15]
```diff
@@ -19,6 +19,10 @@
...
+  `release_plan_admission` files from the results of the `collect-data` task.
```
Comment (johnbieren) [inline pipelines/rhtap-service-push/README.md]: .json

**Review by johnbieren — CHANGES_REQUESTED** [review]

File: pipelines/e2e/README.md (hunk 1 above)
Comment (happybhati) [reply pipelines/e2e/README.md]: Fixed, thanks!

File: pipelines/deploy-release/README.md (hunk 2 above)
Comment (happybhati) [reply pipelines/deploy-release/README.md]: Fixed, thanks!

File: pipelines/e2e/README.md (hunk 3 above)
Comment (happybhati) [reply pipelines/e2e/README.md]: Fixed, thanks!

File: pipelines/push-to-external-registry/README.md (hunk 4 above)
Comment (happybhati) [reply pipelines/push-to-external-registry/README.md]: Fixed, thanks!

File: pipelines/push-to-external-registry/README.md (hunk 5 above)
Comment (happybhati) [reply pipelines/push-to-external-registry/README.md]: My bad, fixed, thanks!

File: pipelines/release-to-github/README.md (hunk 6 above)
Comment (happybhati) [reply pipelines/release-to-github/README.md]: Fixed, thanks!

File: pipelines/release-to-github/README.md (hunk 7 above)
Comment (happybhati) [reply pipelines/release-to-github/README.md]: Fixed, thanks!

File: pipelines/rh-push-to-external-registry/README.md (hunk 8 above)
Comment (happybhati) [reply pipelines/rh-push-to-external-registry/README.md]: Fixed, thanks!

File: pipelines/rh-push-to-external-registry/README.md (hunk 9 above)
Comment (happybhati) [reply pipelines/rh-push-to-external-registry/README.md]: Fixed, thanks!

File: pipelines/rh-push-to-registry-redhat-io/README.md (hunk 10 above)
Comment (happybhati) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: Fixed, thanks!

File: pipelines/rh-push-to-registry-redhat-io/README.md (hunk 11 above)
Comment (happybhati) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: Fixed, thanks!

File: pipelines/rh-push-to-registry-redhat-io/README.md (hunk 12 above)
Comment (happybhati) [reply pipelines/rh-push-to-registry-redhat-io/README.md]: Fixed, thanks!

File: pipelines/rhtap-service-push/README.md (hunk 13 above)
Comment (happybhati) [reply pipelines/rhtap-service-push/README.md]: Fixed, thanks!

File: pipelines/rhtap-service-push/README.md (hunk 14 above)
Comment (happybhati) [reply pipelines/rhtap-service-push/README.md]: Fixed, thanks!

File: pipelines/rhtap-service-push/README.md (hunk 15 above)
Comment (happybhati) [reply pipelines/rhtap-service-push/README.md]: Fixed, thanks!

File: pipelines/deploy-release/README.md [hunk 16]
```diff
@@ -16,6 +16,10 @@ Tekton pipeline to verify Snapshot prior to Deployment
 | taskGitUrl | The url to the git repo where the release-service-catalog tasks to be used are stored | Yes | https://github.com/redhat-appstudio/release-service-catalog.git |
 | taskGitRevision | The revision in the taskGitUrl repo to be used | No | - |
 
+### Changes in 0.14.0
```
Comment (johnbieren) [inline pipelines/deploy-release/README.md]: One too many #

**Review by johnbieren — CHANGES_REQUESTED** [review]
One last thing to fix

File: pipelines/deploy-release/README.md (hunk 16 above)
Comment (happybhati) [reply pipelines/deploy-release/README.md]: Done, thanks :)

**Review by johnbieren — DISMISSED** [review]

**Review by mmalina — DISMISSED** [review]

**Comment by happybhati** [comment]
I did a rebase to resolve the conflict. It needs approval, please. 
CC: @johnbieren @mmalina 
Thanks!

**Review by johnbieren — APPROVED** [review]

**Review by mmalina — APPROVED** [review]

**Comment by happybhati** [comment]
@mmalina or @johnbieren, could you please approve again? I just did a rebase!

**Comment by mmalina** [comment]
> @mmalina or @johnbieren, could you please approve again? I just did a rebase!

The PR is still approved according to Github. The lgtm label is an Openshift CI thing and it would probably only be relevant if we had the Tide check enabled. But the e2e check is passing right now, so it's still good to merge.

**Comment by johnbieren** [comment]
> > @mmalina or @johnbieren, could you please approve again? I just did a rebase!
> 
> The PR is still approved according to Github. The lgtm label is an Openshift CI thing and it would probably only be relevant if we had the Tide check enabled. But the e2e check is passing right now, so it's still good to merge.

Yeah the github approval stays for rebases but goes away for new changes AFAICT, which is good behavior imo

**Pull request merged**